### ISSUE TITLE
A doc reference to a name in another library which is not imported is not linked

### ISSUE THREAD

**Issue by sethladd** [dart-lang/dartdoc#508]
Consider this case, where both of these libraries are in the same package:

```
library foo;

int topLevelName = 1;
```

and

```
library bar;

/// Try to link to [topLevelName]
class Bar {

}
```

Currently, dartdoc will fail to link `topLevelName` when generating the docs for the `Bar` class.

Should it?

**Comment by sethladd** [comment]
@keertip

**Comment by sethladd** [comment]
How would dartdoc uniquely identify the name of topLevelName ?

**Comment by sethladd** [comment]
Note: looks like the current dartdoc-viewer supports this scenario. You can make a link to Future from dart:io, using only `[Future]`

**Comment by sethladd** [comment]
I saw an example in grinder's code that used `[libName.className]`. I don't know how standard that is.

**Comment by zoechi** [comment]
This would allow to reference identifiers which are not made available in the current scope by imports?

**Comment by sethladd** [comment]
I'm not sure what we can or should allow yet :)  It might make sense to figure out a way to reference any name in the current dartdoc process (which would allow you to reference any name in any library that dartdoc knows about as it's processing your package).

**Comment by sethladd** [comment]
So this is what the spec says:

"The scope of a documentation comment always excludes the imported
namespace of the enclosing library. Only names declared in the enclosing library
are considered in scope within a documentation comment."

So, this sounds like `[libName.thingName]` is now allowed.

@seaneagan @devoncarew I saw that grinder has a comment with `[libName.className]` in it. I think that's not a valid doc comment name. Did that ever work?

**Comment by keertip** [comment]
@sethladd, `dartdoc` uses the `analyzer` to resolve the comment references. So if the analyzer cannot resolve the reference, `dartdoc` cannot link to the it.

**Comment by devoncarew** [comment]
My intuition says that:

```
import 'foo.dart' as foo_lib;

/// I like the class [foo_lib.MyClass].
```

should work, but haven't confirmed.

**Comment by keertip** [comment]
Yes, it should work as long as `foo.dart` is all part of the public api.

**Comment by sethladd** [comment]
@keertip how does the analyzer interpret [name] ? What formats can it interpret?  (ie, how does it resolve the names?)

**Comment by sethladd** [comment]
> should work, but haven't confirmed.

@devoncarew I know that's your intuition, but that doesn't sound like what the spec is saying. Or maybe I'm misinterpreting the spec?

**Comment by sethladd** [comment]
We talked about this offline.

@sethladd will talk to Gilad about the spec.

@keertip will investigate exactly what the analyzer does and what we can rely on when resolving names in brackets.

We will not support linking to a name unless that name is imported into the current library.

We opened an issue to hint to the developer when an unknown name is encountered (https://github.com/dart-lang/dartdoc/issues/515)
